### PR TITLE
raftstore: panic when getting value encouters an error

### PR DESCRIPTION
**Pull request by Connor1996**
###  What have you changed?
print error and panic when getting value encounters an error  

###  What is the type of the changes?
- Engineering (engineering change which doesn't change any feature or fix any issue)

###  How is the PR tested?
- No code

###  Does this PR affect documentation (docs) or should it be mentioned in the release notes?
No

###  Does this PR affect `tidb-ansible`?
No

### CODE REVIEW

File: src/raftstore/coprocessor/properties.rs [hunk 1]
```diff
@@ -520,7 +520,7 @@ impl RangeProperties {
     pub fn decode<T: DecodeProperties>(props: &T) -> Result<RangeProperties> {
         match RangeProperties::decode_from_range_properties(props) {
             Ok(res) => return Ok(res),
-            Err(e) => debug!("old_version:decode to RangeProperties failed with err: {:?}, try to decode to SizeProperties", e),
+            Err(e) => warn!("decode to RangeProperties failed with err: {:?}, try to decode to SizeProperties, maybe upgrade from v2.0 or older version?", e),
```
Comment (zhangjinpeng1987) [inline src/raftstore/coprocessor/properties.rs]: How about use info since it is expected behavior.

File: src/storage/kv/mod.rs [hunk 2]
```diff
@@ -646,10 +646,15 @@ impl<I: Iterator> Cursor<I> {
...
             if let Err(e) = self.iter.status() {
-                CRITICAL_ERROR.with_label_values(&["rocksdb"]).inc();
+                CRITICAL_ERROR.with_label_values(&["rocksdb iter"]).inc();
                 if panic_when_unexpected_key_or_data() {
                     set_panic_mark();
                     panic!("Rocksdb error: {}", e);
+                } else {
+                    error!(
+                        "encounter Rocksdb error when iterating";
+                        "err" => %e,
```
Comment (overvenus) [inline src/storage/kv/mod.rs]: Prefer "error". Also could you print the current key, `self.min_key` and `self.max_key`?

Comment (Connor1996) [reply src/storage/kv/mod.rs]: can't print `the current key`, cause cursor already moved on

File: src/storage/kv/mod.rs [hunk 3]
```diff
@@ -646,10 +646,22 @@ impl<I: Iterator> Cursor<I> {
     pub fn valid(&self) -> Result<bool> {
         if !self.iter.valid() {
             if let Err(e) = self.iter.status() {
-                CRITICAL_ERROR.with_label_values(&["rocksdb"]).inc();
+                CRITICAL_ERROR.with_label_values(&["rocksdb iter"]).inc();
                 if panic_when_unexpected_key_or_data() {
                     set_panic_mark();
-                    panic!("Rocksdb error: {}", e);
+                    panic!(
```
Comment (BusyJay) [inline src/storage/kv/mod.rs]: Why panic?

Comment (Connor1996) [reply src/storage/kv/mod.rs]: the error is unexpected and by default `panic_when_unexpected_key_or_data` will return false

File: src/raftstore/store/region_snapshot.rs [hunk 4]
```diff
@@ -139,7 +139,26 @@ impl Peekable for RegionSnapshot {
             self.region.get_end_key(),
         )?;
         let data_key = keys::data_key(key);
-        self.snap.get_value(&data_key)
+        self.snap.get_value(&data_key).map_err(|e| {
+            CRITICAL_ERROR.with_label_values(&["rocksdb get"]).inc();
+            if panic_when_unexpected_key_or_data() {
```
Comment (siddontang) [inline src/raftstore/store/region_snapshot.rs]: can we abstract these to a function? too many places use the same logic.

**Pull request merged**